src/game/client/water/w3dwatertracks.cpp

```diff
@@ -13,15 +13,322 @@
 WaterTracksRenderSystem *g_theWaterTracksRenderSystem;
 #endif
 
+static bool s_pauseWaves;
+static WaveInfo s_waveTypeInfo[6] = { { 28.0f, 18.0f, 25.0f, 0.018f, 900, 0.01f, 0.18f, 1500, 0, "wave256.tga", "Pond" },
+    { 55.0f, 36.0f, 80.0f, 0.015f, 2000, 0.5f, 0.18f, 1000, 6267, "wave256.tga", "Ocean" },
+    { 55.0f, 36.0f, 80.0f, 0.015f, 2000, 0.05f, 0.18f, 1000, 6267, "wave256.tga", "Close Ocean" },
+    { 55.0f, 36.0f, 80.0f, 0.015f, 4000, 0.01f, 0.18f, 2000, 6267, "wave256.tga", "Close Ocean Double" },
+    { 55.0f, 27.0f, 80.0f, 0.015f, 2000, 0.01f, 8.0f, 2000, 5367, "wave256.tga", "Radial" },
+    { 0.0f, 0.0f, 0.0f, 0.0f, 0, 0.0f, 0.0f, 0, 0, nullptr, nullptr } };
+
+WaterTracksObj::WaterTracksObj() : m_stageZeroTexture(nullptr), m_bound(false), m_timeOffsetSecondWave(0) {}
+
+WaterTracksObj ::~WaterTracksObj()
+{
+    Free_Water_Tracks_Resources();
+}
+
+void WaterTracksObj::Render() {}
+
+void WaterTracksObj::Get_Obj_Space_Bounding_Sphere(SphereClass &sphere) const
+{
+    sphere = m_boundingSphere;
+}
+
+void WaterTracksObj::Get_Obj_Space_Bounding_Box(AABoxClass &box) const
+{
+    box = m_boundingBox;
+}
+
+int WaterTracksObj::Free_Water_Tracks_Resources()
+{
+    Ref_Ptr_Release(m_stageZeroTexture);
+    return 0;
+}
+
+void WaterTracksObj::Init(float width, Vector2 const &start, Vector2 const &end, char *texture)
+{
+    Free_Water_Tracks_Resources();
+    m_boundingSphere.Init(Vector3(0.0f, 0.0f, 0.0f), 400.0f);
+    m_boundingBox.Init(Vector3(0.0f, 0.0f, 0.0f), Vector3(400.0f, 400.0f, 1.0f));
+    m_perpDir = end - start;
+    m_startPos = start + m_perpDir * 0.5f;
+    m_perpDir *= 1.0f / m_perpDir.Length();
+    m_waveDir = m_perpDir;
+    m_waveDir.Rotate(1.5707964f);
+    m_startPos -= m_waveDir * width;
+    m_waveDir *= 13.0f;
+    m_startPos -= m_waveDir;
+
+    m_x = 2;
+    m_y = 2;
+    m_elapsedMs = 0;
+    m_velocity = 0.01f;
+    m_totalMs = m_waveDir.Length() / m_velocity;
+    m_fadeMs = 3000;
+    m_stageZeroTexture = W3DAssetManager::Get_Instance()->Get_Texture(texture);
+}
+
+void WaterTracksObj::Init(
+    float width, float length, Vector2 const &start, Vector2 const &end, char *texture, int time_offset)
+{
+    Free_Water_Tracks_Resources();
+    m_initialStart = start;
+    m_initialEnd = end;
+    m_timeOffsetSecondWave = time_offset;
+    m_boundingSphere.Init(Vector3(0.0f, 0.0f, 0.0f), 400.0f);
+    m_boundingBox.Init(Vector3(0.0f, 0.0f, 0.0f), Vector3(400.0f, 400.0f, 1.0f));
+    m_x = 2;
+    m_y = 2;
+
+    m_elapsedMs = m_timeOffsetSecondWave;
+    m_startPos = start;
+    m_waveDir = end - start;
+    m_perpDir = m_waveDir;
+    m_perpDir.Rotate(-1.5707964f);
+    m_perpDir.Normalize();
+    m_waveDir = m_perpDir;
+    m_waveDir.Rotate(1.5707964f);
+
+    m_distanceFromShore = s_waveTypeInfo[m_type].distance_from_shore;
+    m_waveDir *= m_distanceFromShore;
+    m_startPos -= m_waveDir;
+    m_velocity = s_waveTypeInfo[m_type].initial_velocity;
+    m_totalMs = m_distanceFromShore / m_velocity;

```

Comment:
seems to need to be  m_totalMs = (int)(m_distanceFromShore / m_velocity);

Reply:
fixed.